osu.Game/Screens/Ranking/ScorePanelList.cs

```diff
@@ -145,15 +162,92 @@ protected override void Update()
             flow.Padding = new MarginPadding { Horizontal = offset };
         }
 
-        private class Flow : FillFlowContainer<ScorePanel>
+        private bool handleInput = true;
+
+        /// <summary>
+        /// Whether this <see cref="ScorePanelList"/> or any of the <see cref="ScorePanel"/>s contained should handle scroll or click input.
+        /// Setting to <c>false</c> will also hide the scrollbar.
+        /// </summary>
+        public bool HandleInput
+        {
+            get => handleInput;
+            set
+            {
+                handleInput = value;
+                scroll.ScrollbarVisible = value;
+            }
+        }
+
+        public override bool PropagatePositionalInputSubTree => HandleInput && base.PropagatePositionalInputSubTree;
+
+        public override bool PropagateNonPositionalInputSubTree => HandleInput && base.PropagateNonPositionalInputSubTree;
+
+        /// <summary>
+        /// Enumerates all <see cref="ScorePanel"/>s contained in this <see cref="ScorePanelList"/>.
+        /// </summary>
+        /// <returns></returns>
+        public IEnumerable<ScorePanel> GetScorePanels() => flow.Select(t => t.Panel);
+
+        /// <summary>
+        /// Finds the <see cref="ScorePanel"/> corresponding to a <see cref="ScoreInfo"/>.
+        /// </summary>
+        /// <param name="score">The <see cref="ScoreInfo"/> to find the corresponding <see cref="ScorePanel"/> for.</param>
+        /// <returns>The <see cref="ScorePanel"/>.</returns>
+        public ScorePanel GetPanelForScore(ScoreInfo score) => flow.Single(t => t.Panel.Score == score).Panel;
+
+        /// <summary>
+        /// Detaches a <see cref="ScorePanel"/> from its <see cref="ScorePanelTrackingContainer"/>, allowing the panel to be moved elsewhere in the hierarchy.
+        /// </summary>
+        /// <param name="panel">The <see cref="ScorePanel"/> to detach.</param>
+        /// <exception cref="InvalidOperationException">If <paramref name="panel"/> is not a part of this <see cref="ScorePanelList"/>.</exception>
+        public void Detach(ScorePanel panel)
+        {
+            var container = flow.FirstOrDefault(t => t.Panel == panel);
```

Comment:
usual nit (places earlier even used `Single` too)

```suggestion
            var container = flow.SingleOrDefault(t => t.Panel == panel);
```

also goes for `Attach`